dsd/constraints.py

```diff
@@ -576,6 +576,27 @@ class DomainPool(JSONSerializable):
 
     _idx: int = field(compare=False, hash=False, default=0, repr=False)
 
+    find_neighbor_sequences: bool = False
+
+    hamming_probability: Optional[Dict[int, float]] = field(default_factory=dict)
+    """
+    Dictionary that specifies probability of taking a new sequence from the pool that is some integer 
+    number of bases different. 
+    """
+
+    adjacency_list: Dict[str, List[str]] = field(default_factory=dict)
+    """
+    Dictionary storing used sequences and their neighbor sequences a certain Hamming distance away.
+    """
+
+    def __post_init__(self):
+        if not self.hamming_probability: # sets default probability distribution if the user does not
```

Comment:
I don't like using Python "truthiness" like this. I prefer explicit checks. Part of the problem is that both `hamming_probability == None` and `hamming_probability == ` the empty dict evaluate to False. I think it's better if the checks are explicit about what they are checking for, e.g., `if len(self.hamming_probability) == 0`.